### PR TITLE
EES-3172 Fix and refactor get_data_block_responses.py

### PR DESCRIPTION
**Pull request by markhiveit**
This PR fixes a bug in `get_data_block_responses.py` where if a request timed out, it would return the response of the previously processed datablock.

The script has also been refactored.

### CODE REVIEW

File: useful-scripts/get_data_block_responses/get_data_block_responses.py [hunk 1]
```diff
@@ -105,72 +120,63 @@ def print_to_console(text):
...
-
-    headers = {
-        'Content-Type': 'application/json'
-    }
+    def write_this_block(status_code, response_time, response_dict):
+        write_block_to_file(
+            block_id=blockId,
+            release_id=releaseId,
+            subject_id=subjectId,
+            request_dict=queryDict,
```
Comment (duncan-at-hiveit) [inline useful-scripts/get_data_block_responses/get_data_block_responses.py:138]: Should queryDict instead be query_dict to fit in with the other variable naming conventions?

File: useful-scripts/get_data_block_responses/get_data_block_responses.py [hunk 2]
```diff
@@ -105,72 +120,63 @@ def print_to_console(text):
...
+            status_code=resp.status_code,
+            response_time=block_time_end - block_time_start,
+            response_dict={'error': f'Failed to process response text, {e}'})
+        continue
 
-    time.sleep(args.sleep_duration)
+    write_this_block(
+        status_code=resp.status_code,
+        response_time=block_time_end - block_time_start,
+        response_dict=jsonResponse)
```
Comment (duncan-at-hiveit) [inline useful-scripts/get_data_block_responses/get_data_block_responses.py:179]: Should jsonResponse be json_response instead?

File: useful-scripts/get_data_block_responses/get_data_block_responses.py [hunk 3]
```diff
@@ -83,20 +81,37 @@ def print_to_console(text):
...
 for datablock in datablocks:
-    if datablock[0] == 'ContentBlockId':
-        continue
-
-    global queryDict
-    global url
+    time.sleep(args.sleep_duration)
 
-    guid = datablock[0]
+    blockId = datablock[0]
```
Comment (duncan-at-hiveit) [inline useful-scripts/get_data_block_responses/get_data_block_responses.py:114]: Should these variables be snake case?

File: useful-scripts/get_data_block_responses/get_data_block_responses.py [hunk 4]
```diff
@@ -105,72 +120,63 @@ def print_to_console(text):
...
-                f'{query}\n'
-                f'response status: {resp.status_code}\n'
-                f'time for response: {block_time_end - block_time_start}\n'
-                f'response:\n{json.dumps(jsonResponse, sort_keys=True, indent=2)}'
-            )
-        print_to_console(f'Successfully processed block {guid} for subject {subjectId}!')
-    except BaseException:
-        print_to_console(f'file.write failed with block {guid} '
-                         f'subject {subjectId}\n{resp.text}')
+        print_to_console(f'json.loads(resp.text) failed with block {blockId} subject {subjectId}, {e}')
```
Comment (luke-h1) [inline useful-scripts/get_data_block_responses/get_data_block_responses.py:169]: Just wondering if `json.loads(resp.text)` is meant to be wrapped in curly brackets (not sure if you're trying to interpolate a value here)?

Comment (markhiveit) [reply useful-scripts/get_data_block_responses/get_data_block_responses.py]: This is intended - I've changed it to text to be clearer.

**Pull request merged**